pkg/cmd/workflow/shared/shared.go

```diff
@@ -0,0 +1,157 @@
+package shared
+
+import (
+	"errors"
+	"fmt"
+	"path"
+	"strings"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/cli/cli/api"
+	"github.com/cli/cli/internal/ghrepo"
+	"github.com/cli/cli/pkg/prompt"
+)
+
+const (
+	Active           WorkflowState = "active"
+	DisabledManually WorkflowState = "disabled_manually"
+)
+
+type WorkflowState string
+
+type Workflow struct {
+	Name  string
+	ID    int
+	Path  string
+	State WorkflowState
+}
+
+type WorkflowsPayload struct {
+	Workflows []Workflow
+}
+
+func (w *Workflow) Disabled() bool {
+	return w.State != Active
+}
+
+func (w *Workflow) Base() string {
+	return path.Base(w.Path)
+}
+
+func GetWorkflows(client *api.Client, repo ghrepo.Interface, limit int) ([]Workflow, error) {
+	perPage := limit
+	page := 1
+	if limit > 100 || limit == 0 {
+		perPage = 100
+	}
+
+	workflows := []Workflow{}
+
+	for {
+		if limit > 0 && len(workflows) == limit {
+			break
+		}
+		var result WorkflowsPayload
+
+		path := fmt.Sprintf("repos/%s/actions/workflows?per_page=%d&page=%d", ghrepo.FullName(repo), perPage, page)
+
+		err := client.REST(repo.RepoHost(), "GET", path, nil, &result)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, workflow := range result.Workflows {
+			workflows = append(workflows, workflow)
+			if limit > 0 && len(workflows) == limit {
+				break
+			}
+		}
+
+		if len(result.Workflows) < perPage {
+			break
+		}
+
+		page++
+	}
+
+	return workflows, nil
+}
+
+func SelectWorkflow(workflows []Workflow, promptMsg string, states []WorkflowState) (*Workflow, error) {
+	filtered := []Workflow{}
+	candidates := []string{}
+	for _, workflow := range workflows {
+		for _, state := range states {
+			if workflow.State == state {
+				filtered = append(filtered, workflow)
+				candidates = append(candidates, fmt.Sprintf("%s (%s)", workflow.Name, workflow.Base()))
+				break
+			}
+		}
+	}
+
+	var selected int
+
+	err := prompt.SurveyAskOne(&survey.Select{
+		Message:  promptMsg,
+		Options:  candidates,
+		PageSize: 15,
+	}, &selected)
+	if err != nil {
+		return nil, err
+	}
+
+	return &filtered[selected], nil
+}
+
+func ResolveWorkflow(client *api.Client, repo ghrepo.Interface, workflowSelector string) ([]Workflow, error) {
+	if workflowSelector == "" {
+		return nil, errors.New("empty workflow selector")
+	}
+
+	workflow, err := getWorkflowByID(client, repo, workflowSelector)
```

Comment:
Is there a way to determine if the workflow selector is an ID or a name, outside of making the API request and checking for an error?